src/utilities/urlExtension.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Returns the file extension of the path component of a URL
+ * @param {string} url
+ * @returns {string} null if no extension found
+ */
+export function getUrlExtension( url ) {
+
+	let parsedUrl;
+	try {
+
+		parsedUrl = new URL( url );
+
+	} catch ( _ ) {
```

Comment:
Nit: If we're not using the caught error then we shouldn't have to include it in the catch statement. It should be enough to just use

```js
try {
  // ...
} catch {
  // ...
}
```

Reply:
Today I learned: [Optional catch binding](https://github.com/tc39/proposal-optional-catch-binding).

Unfortunately it appears this project's eslint config won't parse that syntax. For the line containing
```js
	} catch {
```
I see
```
/(redacted)/3DTilesRendererJS/src/utilities/urlExtension.js
  13:10  error  Parsing error: Unexpected token {
```

Unless you object, I'll keep the `_` discard value for now and leave the eslint investigation/fix for a later PR.

Reply:
Oh okay good to know! I'll take a look at that another time. Thanks for trying it out.